### DISCUSSION TITLE
unable to nextjs app host on hosting raja ovi panel

### DISCUSSION THREAD

**Discussion by ASalman-Malik** [vercel/next.js#43217] [Help]
got these error what to do to fix them?
/tmp/start-55b04ed0.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-590f6f91.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-658ba453.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-dc4b6219.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-1df30668.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-ecdaab79.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-dc4e0c6b.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-5e92e30f.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-df438e79.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-b1e6b9ab.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-219812f3.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-f226f2c3.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied
/tmp/start-14530f57.sh: line 1: /home/quaeretech-in/public_html/node_modules/.bin/next: Permission denied

**Comment by icyJoseph** [comment]
Do you have permissions, as a user, to open ports and write/read from disk in this panel? I remember working with another service that also used `public_html` and there, I could only drop static files, not a server.

**Reply by ASalman-Malik** [reply]
yes i have all read/write/delete permissions.

**Reply by icyJoseph** [reply]
And this ovi panel can run servers, correct? it has Node.js runtime and all of that. Otherwise, I am not sure here, perhaps you need to contact their customer support. It simply looks like something `next` is trying to do is not allowed, perhaps opening a socket or port.

**Reply by ASalman-Malik** [reply]
getting new error:----------------------
warn  - Detected next.config.js, no exported configuration found. https://nextjs.org/docs/messages/empty-configuration
Error: Could not find a production build in the '/home/quaeretech-in/public_html/.next' directory. Try building your app with 'next build' before starting the production server. https://nextjs.org/docs/messages/production-start-no-build-id
    at NextNodeServer.getBuildId (/home/quaeretech-in/public_html/node_modules/next/dist/server/next-server.js:113:23)
    at new Server (/home/quaeretech-in/public_html/node_modules/next/dist/server/base-server.js:92:29)
    at new NextNodeServer (/home/quaeretech-in/public_html/node_modules/next/dist/server/next-server.js:72:9)
    at NextServer.createServer (/home/quaeretech-in/public_html/node_modules/next/dist/server/next.js:111:16)
    at async /home/quaeretech-in/public_html/node_modules/next/dist/server/next.js:123:31

**Reply by icyJoseph** [reply]
I don't understand what is it your are trying to do though...

- > Detected next.config.js, no exported configuration found.

This means you have a `next.config.js` file without `module.exports = /* some config */`

- > Could not find a production build in the '/home/quaeretech-in/public_html/.next' directory

This means that you did not run `next build`, and you are just trying to do `next start`. I suspect the `next build` command is not allowed to write to disk, and it never creates `.next`.

**Reply by ASalman-Malik** [reply]
same next.config.js is working on local and also on same code publised on vercel working fine but not in here

**Reply by icyJoseph** [reply]
Aha, then my wild guess here is that, even tough you've got certain permissions, the script you are running doesn't so it can't read the config, or write the `.next` build folder, etc. Try to contact their customer support, this looks very specific to this platform.

**Reply by ASalman-Malik** [reply]
okay thanks for the help

**Comment by ASalman-Malik** [comment]
resolved.
